src/util/irep2.cpp

```diff
@@ -176,6 +177,54 @@ unsigned int array_type2t::get_width() const
   return num_elems * sub_width;
 }
 
+unsigned int vector_type2t::get_width() const
+{
+  unsigned int sub_width = subtype->get_width();
+
+  const expr2t *elem_size = vector_size.get();
+  const constant_int2t *const_elem_size =
+    dynamic_cast<const constant_int2t *>(elem_size);
+  assert(const_elem_size != nullptr);
+  unsigned long num_elems = const_elem_size->as_ulong();
+
+  return num_elems * sub_width;
+}
+
+expr2tc vector_type2t::distribute_operation(
```

Comment:
Please add some comments here to explain your implementation.

Reply:
I've added for both at the header and at the source